src/main/java/org/folio/rest/impl/HoldingsStorageAPI.java

```diff
@@ -323,9 +328,10 @@ public void putHoldingsStorageHoldingsByHoldingsRecordId(
           postgresClient.get(HOLDINGS_RECORD_TABLE, HoldingsRecord.class, fieldList, cql, true, false,
             reply -> {
               if(reply.succeeded()) {
-                List<HoldingsRecord> itemList = reply.result().getResults();
+                List<HoldingsRecord> holdingsList = reply.result().getResults();
 
-                if (itemList.size() == 1) {
+                if (holdingsList.size() == 1) {
+                  updateItemEffectiveCallNumbersByHoldings(entity, okapiHeaders, vertxContext);
```

Comment:
Should the update of the items happen before the holdings record update has been attempted?

Are these performed in a database transaction?

Reply:
I've changed the holdings update to a transaction and moved the update of effective call numbers so it will only happen after the holdings are updated.